crates/kubelet/src/kubelet.rs

```diff
@@ -63,8 +69,53 @@ impl<T: 'static + Provider + Sync + Send> Kubelet<T> {
             }
         });
 
-        // This informer listens for pod events.
-        let provider = self.provider.clone();
+        // TODO: How should we configure this value? We should eventually have a max pods setting
+        // just like a normal kubelet, so maybe that?
+        let (error_sender, mut error_receiver) = mpsc::channel::<(KubePod, anyhow::Error)>(200);
+        let client_clone = client.clone();
+        let error_handler = tokio::task::spawn(async move {
+            let client = client_clone;
+            while let Some(e) = error_receiver.recv().await {
+                let (pod, err) = e;
+                let json_status = serde_json::json!(
+                    {
+                        "metadata": {
+                            "resourceVersion": "",
+                        },
+                        "status": {
+                            "phase": Phase::Failed,
+                            "message": format!("{}", err),
+                        }
+                    }
+                );
+
+                debug!(
+                    "Setting pod status for {} using {:?}",
+                    pod.name(),
+                    json_status
+                );
+
+                let data = serde_json::to_vec(&json_status).expect("Should always serialize");
+                let pod_resource =
+                    Resource::namespaced::<KubePod>(&pod.namespace().unwrap_or_default());
+                let pod_name = pod.name();
+                match pod_resource.patch_status(&pod_name, &PatchParams::default(), data) {
+                    Ok(req) => {
+                        if let Err(e) = client.request::<KubePod>(req).await {
+                            error!(
```

Comment:
Is just logging an error the right thing to do here?

Reply:
We don't want to crash things just because of a single error. We could try wrapping this in the retry stuff you wrote though